### PR TITLE
Add queue for code indexer

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

**Comment by lunny** [comment]
blocked by #10133.

**Comment by lafriks** [comment]
@lunny ping

File: modules/indexer/code/indexer.go [hunk 1]
```diff
@@ -127,3 +174,79 @@ func Init() {
...
+func DeleteRepoFromIndexer(repo *models.Repository) {
+	indexData := &IndexerData{RepoID: repo.ID, IsDelete: true}
+	if err := indexerQueue.Push(indexData); err != nil {
+		log.Error("Delete repo index data %v failed: %v", indexData, err)
+	}
+}
+
+// UpdateRepoIndexer update a repository's entries in the indexer
+func UpdateRepoIndexer(repo *models.Repository) {
+	indexData := &IndexerData{RepoID: repo.ID, IsDelete: false}
```
Comment (lafriks) [inline modules/indexer/code/indexer.go]:
```suggestion
	indexData := &IndexerData{RepoID: repo.ID}
```

File: modules/indexer/code/indexer.go [hunk 2]
```diff
@@ -127,3 +174,79 @@ func Init() {
...
+			break
+		}
+		for _, id := range ids {
+			select {
+			case <-isShutdown:
+				log.Info("Repository Indexer population shutdown before completion")
+				return
+			default:
+			}
+			if err := indexerQueue.Push(&IndexerData{RepoID: id, IsDelete: false}); err != nil {
```
Comment (lafriks) [inline modules/indexer/code/indexer.go]:
```suggestion
			if err := indexerQueue.Push(&IndexerData{RepoID: id}); err != nil {
```

**Comment by lunny** [comment]
@lafriks  done.

**Comment by lafriks** [comment]
Tests seem to be broken

**Comment by lunny** [comment]
It's ready to review.

File: integrations/mssql.ini.tmpl [hunk 3]
```diff
@@ -14,6 +14,9 @@ ISSUE_INDEXER_PATH = integrations/indexers-mssql/issues.bleve
 REPO_INDEXER_ENABLED = true
 REPO_INDEXER_PATH = integrations/indexers-mssql/repos.bleve
 
+[queue.code_indexer]
+TYPE = noqueue
```
Comment (zeripath) [inline integrations/mssql.ini.tmpl]: It might be better to call this queue type `immediate`

**Review by zeripath — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by lunny** [comment]
Make L-G-T-M work

**Pull request merged**